zul/src/main/java/org/zkoss/zul/Tree.java

```diff
@@ -986,6 +1021,53 @@ public void setMultiple(boolean multiple) {
 		}
 	}
 
+	/** Returns whether tristate mode is activated.
+	 * At the same time, setCheckmark(true) and setMultiple(true) will be called automatically.
+	 * Once this mode is on, setCheckmark(false) and setMultiple(false) will also set tristate to false.
+	 * <p>Default: false.
+	 * @since 10.0.0
+	 */
+	public boolean isTristate() {
+		return _tristate;
+	}
+
+	/** Sets whether tristate mode is activated.
+	 * At the same time, setCheckmark(true) and setMultiple(true) will be called automatically.
+	 * Once this mode is on, setCheckmark(false) and setMultiple(false) will also set tristate to false.
+	 * @since 10.0.0
+	 */
+	public void setTristate(boolean tristate) {
+		if (getEngine() instanceof EmptyTreeEngine)
+			return;
+		if (_tristate != tristate) {
+			_tristate = tristate;
```

Comment:
This state should be updated as well no mater it's EmptyTreeEngine or others.

Reply:
so even if users who don't have zkmax module, they also can change the state of "tristate", but got nothing change actually?

Reply:
I currently use `tristate == false` to control some behaviour